### DISCUSSION TITLE
Release 1.4: Common tasks, Dev acknowledgement, versioning, helper packages/websites, etc

### DISCUSSION THREAD

**Discussion by manciniedoardo** [pharmaverse/admiral#2394] [General]
[Master PR](https://github.com/pharmaverse/admiral/pull/2919)

# Common Release Tasks

All of these small Release Tasks should be done in one issue and not split out across multiple issues and PRs.  I have created a common task list for us to check and hope to develop the task (if needed) below.  If we can standardized this stuff, then we could build a `release_function` that is a wrapper around the https://usethis.r-lib.org/reference/use_release_issue.html

## Prepare for release:
* [x] New devs to description file and old devs to acknowledgements
  - Moved to acknowledgments: Junze Zhang, G Gayatri, Daniel Sjoberg, Dinakar Kulkarni, Solveig Holmgaard
  - New authors: @MattBearham - let me know if you want an ORCID added too, Matt
* [x] Review FAQs 
* [x] Review [Cheatsheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf) https://github.com/pharmaverse/admiral/issues/2758 - as per feedback from last process, doing this in a separate issue as we wish tomerge it right before 1.3 release.
* [x] Polish news for admiral and admiraldev
* [x] Update README schedule for future releases
* [x] Review pharmaverse admiral youtube videos (https://www.youtube.com/channel/UCxQFEv8HNqM01DXzdQLCy6Q)
* [x] Check Size of package ( Code [here](https://github.com/pharmaverse/admiral/discussions/2394#discussioncomment-13550187) )
* [x] Check version numbers are correct in DESCRIPTION (i.e. admiraldev, pharmaversesdtm, others?)
* [ ] Check Remotes is removed form DESCRIPTION
* [ ] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_admiral.html)
* [x] Scan and remove non-ascii characters - https://github.com/pharmaverse/admiral/issues/2456
* [x] Check if any deprecation processes should be advanced, as described in [Gradual deprecation](https://lifecycle.r-lib.org/articles/communicate.html#gradual-deprecation)
* [x] `urlchecker::url_check()` checked as of 19th June
![image](https://github.com/user-attachments/assets/d3c091b6-7373-4347-801a-0698242a2bee)
* [x] `devtools::check(remote = TRUE, manual = TRUE)` output as of 19th June:
![image](https://github.com/user-attachments/assets/be2ee109-857b-4d27-9feb-24a09e3a497b)
* [ ] `devtools::check_win_devel()` - doesn't work with Remtoes
* [ ] Run roche templates
* [ ]  Use Insight Engineering [reverse dependency checks](https://github.com/pharmaverse/admiral/actions/workflows/verdepcheck.yml). Requires manually running the Action.  Might want to check the Rust one published on pharmaverse slack https://nanx.me/blog/post/revdep-speedrun/
* [ ] Review pharmaverse examples for impact (use News to assess)
* [ ] Draft blog post

## Submit to CRAN:
* [ ] Be sure to use [skip vbump] in PR Title to prevent upversioning
* [ ] `usethis::use_version('major')`
* [ ] `devtools::submit_cran()`
* [ ] Approve email
* [ ] Monitor submission form here: https://r-hub.github.io/cransays/articles/dashboard.html
* [ ] Start using CRAN Comments, `usethis::??`

## CRAN Accepted:
* [ ] `usethis::use_github_release()`
* [ ] `usethis::use_dev_version()`
* [ ] Finish blog post
* [ ] Add Releases and tar.gz file to the Release
* [ ] Review admiraldiscovery (presentations and find a function)
* [ ] Push forward deprecation 

### New presentations
- [ ] Posit Conference
- [ ] Something from Edoardo?
- [ ] Something from Fanny?

https://github.com/pharmaverse/admiraldiscovery/pull/64

### News devs to description File and old devs to acknowledgements - NB: check admiraldev as well 

https://github.com/pharmaverse/admiral/pull/2748
https://github.com/pharmaverse/admiraldev/pull/492

- [x] Gayatri
- [x] Junze
- [x] Arianna

### Devs to moved to acknowledgments (not actively working on admiral anymore) at the bottom of README:
- Dinakar
- Junze
- Solveig

#### New acknowledgements
- [ ] Check closed PRs for mini-PRs and issues raised by users

## Review FAQs 
- [ ] Check for outdated info
- [ ] Chances to consolidate information or remove
- [ ] Possible to link to other information to help minimize upkeep

## Polish News
- [ ] Each news item should have a linked issue
- [ ] Check for consistency and grammar

**Comment by bms63** [comment]
I'm going to move this to a Discussion as we have a bunch of release tasks that should be compiled into one place.

**Comment by bms63** [comment]
Hi all, @pharmaverse/admiral 

I think building a Discussion around Release is needed.  There are a lot of small tasks that get separate issues/PRs that I think should be done all in one PR.  Sometimes we even forget to do them.

Do you see any other tasks that should be set up for a release?  Some of these are form the `usethis` package and others are ones that we do specific to admiral.  Please drop anything that needs to be added into this discussion and I will add in

**Reply by ddsjoberg** [reply]
Very good idea!

**Reply by manciniedoardo** [reply]
I've done the conference and contributors updates and linked the corresponding PRs - if anyone does any other bits please use the same branch and PR

**Comment by bms63** [comment]


**Reply by bms63** [reply]
Actual paper 

https://www.[pharmasug.org/proceedings/2024/ST/PharmaSUG-2024-ST-461.pdf](https://www.pharmasug.org/proceedings/2024/ST/PharmaSUG-2024-ST-461.pdf)

**Comment by bms63** [comment]
Cool Issue Template - https://github.com/insightsengineering/teal/issues/new?assignees=KlaudiaBB%2Ccicdguy&labels=release&projects=&template=release.yml&title=%5BRelease%5D%3A+%3Cversion%3E

**Comment by bms63** [comment]
Making note here on cheatsheet - needs a little more thought on when to update https://github.com/pharmaverse/admiral/pull/2600

**Comment by bms63** [comment]
I don't know what this means? @manciniedoardo 
![image](https://github.com/user-attachments/assets/45ad188d-fc46-4502-adc6-b5fc26d52d6f)

**Comment by ddsjoberg** [comment]
@bms63 I wouldn't worry about the qpdf and latex warnings. If you want, you can install it on your machine to avoid the message. Resolving the latex warning will depend on how the latex distribution on your machine, so I would just ignore it. These both arise from building the PDF manual, which CRAN does on their end anyway.

Has the size of the package grown since the last release? Unless the rules have changed, you may be auto-rejected because the build is >5MB. But I know there are packages that exceed this size on CRAN, but I have never gone through that process. 🤷🏼

**Reply by bms63** [reply]
Thanks @ddsjoberg!!!  I was more poking at @manciniedoardo about what we are doing about these warnings and notes - we should be all clean.  

* I have a linux installation with qpdf installed so this one doesn't come up for me.
* when i do a manual check on the size for latest admiral dev I get this 
![image](https://github.com/user-attachments/assets/02fa9191-f939-487d-9417-e8d1c8a1ae9c)
* Not sure on the tidy Note ?
* @manciniedoardo maybe add to `.Rbuildignore` the `admiral-manual.tex` so it doesn't show up as a Note?

**Comment by bms63** [comment]
## Sniff out unicode characters in .Rd files

search_specific_unicode_in_rd_files <- function(folder_path, unicode_char) {
  # List all `.Rd` files in the folder
  rd_files <- list.files(path = folder_path, pattern = "\\.Rd$", full.names = TRUE)
  
  for (file in rd_files) {
    # Read each `.Rd` file
    rd_content <- readLines(file, encoding = "UTF-8")
    
    # Identify lines containing the specific Unicode character
    has_unicode <- grepl(unicode_char, rd_content)
    unicode_lines <- rd_content[has_unicode]
    
    if (length(unicode_lines) > 0) {
      cat("File:", file, "\n")
      cat("Lines with the Unicode character ℹ:\n")
      print(unicode_lines)
      cat("\n")
    }
  }
}

# Define the Unicode character (information source symbol)
unicode_char <- "ℹ"

# Run the function with your folder path
search_specific_unicode_in_rd_files("/cloud/project/man", unicode_char)

**Reply by bundfussr** [reply]
On linux/unix systems you could also use
```
grep --color=auto $(printf \\U2139) *.Rd
```

**Reply by bms63** [reply]
grep forever!!  much simpler :)

**Comment by manciniedoardo** [comment]
@1mgroot @gg106046 @ddsjoberg @cicdguy @SolveigHolmgaard 

This is just a courtesy notification that as per our internal guidelines, for the 1.4 release we will be moving your names from the authors section of the DESCRIPTION file to the acknowledgements section of the README, because you have not been involved with the package during the 1.3 release cycle. Of course, if in future your involvement with admiral picks up again, we'll be happy to re-add you as authors.

**Comment by bms63** [comment]
Quick note on 404 to check 

https://github.com/pharmaverse/admiral/issues/2948

**Comment by bms63** [comment]
AZ presentation - https://pharmaverse.github.io/az-pharmaverse-presentation/#/title-slide
